lib/eex/lib/eex/compiler.ex

```diff
@@ -478,4 +517,47 @@ defmodule EEx.Compiler do
     # length(~c"<%") == 2
     column + 2 + length(mark)
   end
+
+  defp code_snippet(source, meta, arrow_padding) do
+    line_start = max(meta.line - 3, 1)
+    line_end = meta.line
+
+    case String.split(source, ["\r\n", "\n"]) do
+      lines when length(lines) < line_start ->
```

Comment:
I'm not 100% if this make sense, but I had to do it to make this test pass: 

https://github.com/elixir-lang/elixir/blob/86f2d5096c976f9cb0a322d54c62d2f1752dd232/lib/eex/test/eex_test.exs#L647